indexing-hadoop/src/main/java/io/druid/indexer/HadoopDruidIndexerMapper.java

```diff
@@ -104,6 +105,10 @@ public final static InputRow parseInputRow(Object value, InputRowParser parser)
     if(parser instanceof StringInputRowParser && value instanceof Text) {
       //Note: This is to ensure backward compatibility with 0.7.0 and before
       return ((StringInputRowParser) parser).parse(value.toString());
+    } else if(parser instanceof StringInputRowParser && value instanceof BytesWritable) {
```

Comment:
I can see it going both ways. But, only reason I added Text as special case was to maintain backward compatibility. In general the "contract" I believe we want to have is that InputFormat can return anything and specific InputRowParser configured should be able to handle that "anything" and be able to give an InputRow from that.

I can see your point of view about adding special handling for "popular" hadoop InputFormat return types which are String-ish. But, at the same time, it is possible to write BytesWritableInputRowParser (that probably just extends StringInputRowParser) and use that. That can be added as an extension( which might contain all other StringishTypeInputRowParser implementations)  or in core code.